.github/workflows/ci.yaml

```diff
@@ -28,51 +32,50 @@ jobs:
           - stm32mp1
           - stm32wl
           - stm32wb
+
     env:
       CRATES: ${{ matrix.crate }}
       CARGO_INCREMENTAL: 0
+      SVDTOOLS: svdtools
+
     steps:
       - name: Checkout code
         uses: actions/checkout@v2
+
       - name: Install Rust
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
           profile: minimal
           override: true
           components: rustfmt
+
       - name: Install xmllint
         run: sudo apt-get install libxml2-utils
-      - name: Install Python dependencies
-        run: |
-          pip3 install --user setuptools wheel
-          pip3 install --user svdtools
-      - name: Put pip binary directory into path
-        run: echo "~/.local/bin" >> $GITHUB_PATH
-      - name: Cache Cargo installed binaries
-        uses: actions/cache@v1
-        id: cache-cargo
-        with:
-          path: ~/cargo-bin
-          key: ${{ runner.os }}-svd2rust-0.21.0
-      - name: Install svd2rust
-        if: steps.cache-cargo.outputs.cache-hit != 'true'
-        uses: actions-rs/install@v0.1
+
+      - name: Create working crate
+        run: make crates
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
         with:
-          crate: svd2rust
-          version: 0.21.0
-      - name: Copy svd2rust to cache directory
-        if: steps.cache-cargo.outputs.cache-hit != 'true'
+          working-directory: ${{ matrix.crate }}
+          key: svdtools-${{ env.SVDTOOLS_VERSION }}-svd2rust-${{ env.SVD2RUST_VERSION }}
+
+      - name: Install tools
         run: |
-          mkdir ~/cargo-bin
-          cp ~/.cargo/bin/svd2rust ~/cargo-bin
-      - name: Put new cargo binary directory into path
-        run: echo "~/cargo-bin" >> $GITHUB_PATH
+          cargo install svdtools --version $SVDTOOLS_VERSION --target-dir deps-target
+          cargo install svd2rust --version $SVD2RUST_VERSION --target-dir deps-target
+          rm -rf deps-target
```

Comment:
Looking good. What's the point of using deps-target and deleting it?

Reply:
1 target dir for rule them all.
Umm. For speed up builds when rebuild both.
Cleaning to make cache smaller.

Reply:
Makes sense, thanks for explaining.

Reply:
Oh. Cleaning doesn't need as it is custom dir now. I've added it early when used `target` for this.